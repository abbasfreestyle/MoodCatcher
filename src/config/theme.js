const palette = {
  white: '#FFFFFF',
  mystic: '#E0E7ED',
  iron: '#D4D4D5',
  trout: '#4A4C58',
  limedSpruce: '#323C48',
  tuna: '#393C46',
  black: '#000000',
  brilliantRose: '#F26AA3',
  buttercup: '#F6B21B',
  red: '#FF0000',
  monza: '#B3001E',
  redDamask: '#D65E49',
  java: '#1EB8C3',
  turquoiseBlue: '#67EEA1',
  wisteria: '#A073B2',
  shakespeare: '#4895D0'
};

const theme = {
  primary: palette.wisteria,
  primaryVariant: palette.trout,
  secondary: palette.brilliantRose,

  background: palette.mystic,
  surface: palette.white,
  error: palette.monza,

  onPrimary: palette.white,
  onPrimaryAlt: palette.white,
  onPrimaryVariant: palette.iron,
  onSecondary: palette.white,
  onBackground: palette.limedSpruce,
  onSurface: palette.limedSpruce,
  onError: palette.white,

  shadow: palette.black,
  disabled: palette.iron
};

export default theme;
