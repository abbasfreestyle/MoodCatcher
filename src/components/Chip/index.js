import React from 'react';
import PropTypes from 'prop-types';

import { Text, View, StyleSheet } from 'react-native';

import theme from 'theme';

const styles = StyleSheet.create({
  container: {
    paddingRight: 5,
    paddingBottom: 5
  },
  chipText: {
    fontSize: 11,
    paddingHorizontal: 10,
    textTransform: 'uppercase'
  },
  chip: {
    borderRadius: 12.5,
    justifyContent: 'center',
    alignItems: 'center',
    height: 25
  }
});

const Chip = props => {
  const { text, backgroundColor } = props;
  return (
    <View style={styles.container}>
      <View style={[styles.chip, { backgroundColor }]}>
        <Text ellipsizeMode="tail" numberOfLines={1} style={styles.chipText}>
          {text}
        </Text>
      </View>
    </View>
  );
};

Chip.defaultProps = {
  backgroundColor: theme.disabled
};

Chip.propTypes = {
  text: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string
};

export default Chip;
