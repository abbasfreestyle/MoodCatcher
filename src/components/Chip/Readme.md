# Chip

This component displays a grey chip with text. It can be customized by optionally making it tappable, display an avatar and/or show a close icon to the right hand side.

## Usage

```js
import { Chip } from 'components';

render() {
  return <Chip text="Chuck Norris" />;
}
```