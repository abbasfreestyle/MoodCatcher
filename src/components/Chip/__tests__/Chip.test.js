import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import theme from 'theme';

import Chip from '..';

const requiredProps = {
  text: 'Happy'
};

const wrapper = shallow(<Chip {...requiredProps} />);

describe('Chip.', () => {
  describe('Snapshots.', () => {
    afterEach(() => {
      wrapper.setProps({ ...Chip.defaultProps, ...requiredProps });
    });

    test('should match default props', () => {
      expect(Chip.defaultProps).toMatchSnapshot();
    });

    test('should match with default props provided', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when a different background color is provided', () => {
      wrapper.setProps({ backgroundColor: theme.surface });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
