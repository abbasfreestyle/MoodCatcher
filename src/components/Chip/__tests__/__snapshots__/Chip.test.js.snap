// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip. Snapshots. should match default props 1`] = `
Object {
  "backgroundColor": "#D4D4D5",
}
`;

exports[`Chip. Snapshots. should match when a different background color is provided 1`] = `
<View
  style={
    Object {
      "paddingBottom": 5,
      "paddingRight": 5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12.5,
          "height": 25,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "fontSize": 11,
          "paddingHorizontal": 10,
          "textTransform": "uppercase",
        }
      }
    >
      Happy
    </Text>
  </View>
</View>
`;

exports[`Chip. Snapshots. should match with default props provided 1`] = `
<View
  style={
    Object {
      "paddingBottom": 5,
      "paddingRight": 5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12.5,
          "height": 25,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#D4D4D5",
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "fontSize": 11,
          "paddingHorizontal": 10,
          "textTransform": "uppercase",
        }
      }
    >
      Happy
    </Text>
  </View>
</View>
`;
